projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts

```diff
@@ -0,0 +1,287 @@
+        declarations: [
+          WriteAddressFieldComponent,
+          WrapperComponent,
+
+        ],
+        providers: [
+          { provide: AddressService, useValue: addressService }]
+      })
+      .compileComponents();
+
+    fixture = TestBed.createComponent(WrapperComponent);
+    wrapperComponent = fixture.componentInstance;
+    wrapperComponent.addressField.patchValue(null);
+
+    debugElement = fixture.debugElement;
+    fixture.detectChanges();
+
+  }));
+
+  it('should render only title, lookup component and manual link when address not set', () => {
+
+    expect(debugElement.query($POSTCODE_LOOKUP)).toBeTruthy();
+    expect(debugElement.query($SELECT_ADDRESS)).toBeFalsy();
+    expect(debugElement.query($MANUAL_LINK)).toBeTruthy();
+
+    /*     expect(debugElement.query($ADDRESS_COMPLEX_FIELD)).toBeTruthy();
+        expect(debugElement.query($ADDRESS_COMPLEX_FIELD).nativeElement['hidden']).toBeTruthy(); */
+
+  });
+
+  /*   it('should render only address lines if field is search ', () => {
+      wrapperComponent.componentUnderTest.isExpanded = true; // false by default
+      fixture.detectChanges();
+      expect(debugElement.query($TITLE).nativeElement.innerHTML).toEqual(CASE_FIELD_LABEL);
+      expect(debugElement.query($POSTCODE_LOOKUP)).toBeFalsy();
+      expect(debugElement.query($SELECT_ADDRESS)).toBeFalsy();
+      expect(debugElement.query($MANUAL_LINK)).toBeFalsy();
+      expect(debugElement.query($ADDRESS_COMPLEX_FIELD)).toBeTruthy();
+      expect(debugElement.query($ADDRESS_COMPLEX_FIELD).nativeElement['hidden']).toBeFalsy();
+    }); */
+
+  /* it('should render only title, lookup component and manual link when writeComplexFieldComponent is null', () => {
+    wrapperComponent.componentUnderTest.writeComplexFieldComponent = null;
+    fixture.detectChanges();
+    expect(debugElement.query($TITLE).nativeElement.innerHTML).toEqual(CASE_FIELD_LABEL);
+    expect(debugElement.query($POSTCODE_LOOKUP)).toBeTruthy();
+    expect(debugElement.query($SELECT_ADDRESS)).toBeFalsy();
+    expect(debugElement.query($MANUAL_LINK)).toBeTruthy();
+
+    expect(debugElement.query($ADDRESS_COMPLEX_FIELD)).toBeTruthy();
+    expect(debugElement.query($ADDRESS_COMPLEX_FIELD).nativeElement['hidden']).toBeTruthy();
+
+  }); */
+
+  it('should render only title, lookup component and address when address set', () => {
+
+    const address = new AddressModel();
+    address.addressLine1 = 'Address Line 1';
+    address.addressLine2 = 'Address Line 2';
+    address.addressLine3 = 'Address Line 3';
+    address.postTown = 'postTown';
+    address.county = 'county';
+    address.postCode = 'postCode';
+    address.country = 'country';
+
+    wrapperComponent.addressField = new FormControl(address);
+    fixture.detectChanges();
+
+    expect(debugElement.query($POSTCODE_LOOKUP)).toBeTruthy();
+    expect(debugElement.query($SELECT_ADDRESS)).toBeFalsy();
+
+    /* expect(debugElement.query($ADDRESS_COMPLEX_FIELD)).toBeTruthy();
```

Comment:
@johnbenjamin-hmcts - as above

Reply:
@connorpgpmcelroy - could you please check and revert?

Reply:
@Daniel-Lam - I have updated the code.